src/magneticField/PolarizedSingleModeMagneticField.cpp

```diff
@@ -0,0 +1,124 @@
+#include "crpropa/magneticField/PolarizedSingleModeMagneticField.h"
+
+namespace crpropa {
+
+PolarizedSingleModeMagneticField::PolarizedSingleModeMagneticField(double B_0, double wavelength, Vector3d r_0, Vector3d e_1, Vector3d e_2) {
+	setB0(B_0);
+	setWavelength(wavelength);
+	setr0(r_0);
+	sete1(e_1);
+	sete2(e_2);
+}
+
+Vector3d PolarizedSingleModeMagneticField::getField(const Vector3d &pos, const double &sigma) const {
+	Vector3d deltar = pos - r_0;
+
+	Vector3d unitVector_1;
+	if (e_1.getR() == 0)
+		throw std::runtime_error("Vector e_1 cannot be zero.");
```

Comment:
For consistency with the rest of CRPropa, exceptions should be handled with kiss:  
KISS_LOG_ERROR, KISS_LOG_WARNING, etc.

Reply:
Is this really the case? I found throw std::runtime_error error handling in most of the source files. Therefore, I would prefer to keep this way, unless there is a big majority against it.

Reply:
In the new verison, there are no warnings, just errors. For the latter I used runtime_errors.